### PR TITLE
gh-117456: Floating-point arithmetic rounding error bug fix in assertAlmostEqual and assertNotAlmostEqual

### PR DESCRIPTION
**Pull request by jm-willy**
# Floating-point arithmetic rounding error bug fix in assertAlmostEqual and assertNotAlmostEqual

```python
#117456:  Addition of operators <= and >= as another way to pass assertAlmostEqual and assertNotAlmostEqual.

# Explanation for assertAlmostEqual 

#Passes 50% of rounding errors
# Half the time fails, the rounding error of diff makes it slightly larger than delta. 
# The other half of the time, the rounding error of diff makes it slightly smaller than delta and passes. 
diff = abs(first - second)
diff <= delta

#Covers the other 50% of rounding errors
# Passes, second +- delta gives no rounding errors is second is int. 
# If it does, half of the time slightly expands the upper and/or lower bounds making it slightly easier to pass.
# The other half fails because the boundaries are narrower than expected.
upper_bound = second + delta
lower_bound = second - delta
first <= upper_bound
first >= lower_bound

# Both cover 100% of errors
```

* Issue: gh-117456

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
thn

**Review by ghost — APPROVED** [review]

File: Lib/unittest/case.py [hunk 1]
```diff
@@ -13,15 +13,20 @@
...
 from . import result
-from .util import (strclass, safe_repr, _count_diff_all_purpose,
-                   _count_diff_hashable, _common_shorten_repr)
+from .util import (
+    strclass,
+    safe_repr,
+    _count_diff_all_purpose,
+    _count_diff_hashable,
+    _common_shorten_repr,
+)
```
Comment (skirpichev) [inline Lib/unittest/case.py:22]: Your pr has a lot of formatting changes.  I suggest you to revert all this stuff, include only changes, relevant to your issue.

File: Lib/unittest/case.py [hunk 2]
```diff
@@ -13,15 +13,20 @@
...
 
 __unittest = True
 
 _subtest_msg_sentinel = object()
 
-DIFF_OMITTED = ('\nDiff is %s characters long. '
-                 'Set self.maxDiff to None to see it.')
+DIFF_OMITTED = "\nDiff is %s characters long. " "Set self.maxDiff to None to see it."
```
Comment (skirpichev) [inline Lib/unittest/case.py:28]: Another example.

File: Lib/unittest/case.py [hunk 3]
```diff
@@ -31,11 +36,13 @@ class SkipTest(Exception):
     instead of raising this directly.
     """
 
+
```
Comment (skirpichev) [inline Lib/unittest/case.py:39]: And another...

File: Lib/unittest/case.py (hunk 1 above)
Comment (JelleZijlstra) [reply Lib/unittest/case.py:22]: It looks like the whole file was formatted with Black. As a maintainer of Black I approve, but as a maintainer of CPython I prefer that PRs are more focused.

**Comment by JelleZijlstra** [comment]
Closing, see the issue.

**Pull request closed without merging**